### ISSUE TITLE
Print edition for JMM

### ISSUE THREAD

**Issue by oscarlevin** [OpenDiscreteMath/ibl-combinatorics#114]
Before the end of the year, I would like to finalize a print edition of the text, partially so that we can give away a few at JMM through AIM's booth and perhaps Mitch's talk.  So what do we still need to do here?

1. We need to pick a size.  This strikes me as more of a workbook, so larger dimensions seem appropriate.  Any thoughts?
2. We need a new prefix.  That is, preface.
3. We need a cover.
4. We need frontmatter.

Anything else?

**Comment by mitchkeller** [comment]
You seem to have read my mind, as I sent @kentemorrison an email about books for the AIM booth mentioning some of this yesterday not long before you made this issue.

1. I like the idea of a bigger size. My book is 7.44" x 9.69", and I've got all the geometry stuff worked out for that size if we're interested in trying to just run with that (but keeping the various header and layout features already loaded). I also think 8x10 could be fine. I have some sort of psychological issue with 8.5x11 being used for a book, but if others think it's a good size, we can go with that. Do we want to get really creative and increase the inter-problem and inter-task spacing to give students space to take a note or write in an answer where things are short, or just go with a larger page size and strive to have more margin space for notes and such? I can work on this and fixing overfull hboxes once we agree on a page size.
2. I'm assuming "prefix" is some sort of autocorrect for "preface". I'll draft something this afternoon.
3. Yes, we do. I'm inclined to go with something rather simple for this first CreateSpace edition. I could do something up in Illustrator later today as a first attempt at getting the visuals right and then we can do it in the proper template once the page count stabilizes after the page size change. My initial inclination is to draw inspiration from the cover of Bogart's traditional standard combinatorics text, which has a dark blue cover with silver or white text.
4. Yes, but I guess that also means more about what front matter items you're thinking we should get included.

I would add:

5. We should decide where the GFDL gets put in the print version. Right now it's a "numbered" appendix after the Index. This looks weird to me in the TOC, and I think it's also annoying to someone who wants to flip to the back of the book to find the index, which they would expect to find at the very back. AATA has it as the first appendix, which I kind of like, but would mess up the appendix numbering. My inclination would be to either put it before the index or between the EGF appendix (C) and the Hints (currently D, but would become E). The latter appeals because then the two backmatter pieces that are of most frequent use are at the very back.
6. I don't know how much coding this would require, but is there some sort of way that we could give indications of the problems that have hints in the back? If one of us came up with an easy way to do this, I would be in favor of it, but I don't want to see us investing lots of time in this.

**Comment by mitchkeller** [comment]
#116 helps out with 2 and 4 and does something with 5.

As far as 1 goes, I see the current PDF is 8.5 by 11. I don't have terribly strong objections to sticking with that if we want to. I may do some work to try making a few PDFs with different page sizes for us to look at, although it's really  hard to decide how something is going to look in print based purely on comparing the PDF pages on screen.

**Comment by oscarlevin** [comment]
For 1, I think trying a 8x10 would be a reasonable first guess.  I agree, that a 8.5x11 looks a little "cheap".  I can work on this today, unless you have a strong desire to do so.  

I agree with your comments about the GFDL.  I'll try to compile your most recent version to see how that looks.

For 6, I agree, this would be really nice.  I don't think it would be that difficult, although I will need an hour or so to get over the inertia of not having thought about xslt for a few months.  Perhaps end each problem with a hint with an "(h)", that hyperlinks to the hint in the pdf version, or just sits there in print.  We could so something fancy like a margin symbol, but we already have those for the problem type.  Or maybe put the H as a "qed" symbol for exercises.  Once I get the xslt code sorted, we can play with presentation.

**Comment by mitchkeller** [comment]
Go ahead and work on the layout if you’ve got time and are in the mood. I’ve got a few things going on this afternoon on short notice, but think I can do something about a cover at least.

Will think about the hint markers and comment more later.

**Comment by kentemorrison** [comment]
I definitely do not like 8 1/2 by 11. I would go with the size of Mitch's book or with 8 by 10.

**Comment by mitchkeller** [comment]
A couple of things I'm noticing now with the boxes in place:

* If a problem has more than one type or has the especially interesting \Rightarrow, then the type indicator sticks out to the left of the box. Initially I tried just changing the -1em to -0.75 em for shifting the margin symbol closer, which solved the problem for just especially interesting. Then I found things with more than one type. (Problem 51 is an example of this.)
* Footnotes that occur inside an activity environment now are being rendered with a letter instead of a number for the marker and they occur inside the box for the problem. I'm not sure that I'm enthralled with this. (See Problem 33.)

Can you give more details about what's bugging you about some of the boxes? Perhaps screenshots of things that could use work?

**Comment by oscarlevin** [comment]
The intended look of the boxes is to have no border on the top or left, a border on the right and bottom, and the shadow on the bottom.  Most of the time when printed this is what shows up, but sometimes there is a very light border on the top or left, and often in a pdf reader these show up.  And that looks bad when you have the problem type markers, as you say.  The other issue is that sometimes the borders  on the right/bottom show up with different thickness (or not at all) at different zoom levels in adobe.

I agree we need to get the footnotes thing sorted too.  I can look into that this weekend.

I'm also not committed to this particular box style, but it would be nice to have something to separate activities from each other and the text.

**Comment by kentemorrison** [comment]
My mind is not so closed on the page size. I looked at an 8 1/2 by 11 book on the AIM list (How We Got From There to Here: A Story of Real Analysis). It has 205 pages, which is about 20 more than Bogart in this size. It would be okay since it would have the workbook feel. In the 8 by 10 size it would be about 210 pages. Still not fat, but the GFDL takes up 10 pages. 

What about not printing the GFDL? I see that Judson's algebra book (published by Lon Mitchell's Orthogonal Publishing) doesn't have the printed GFDL; instead it has a notice on the copyright page referring to the license and giving a URL for it. However, close reading of the license does make it sound like it should be printed with each copy.

But both of your books use a Creative Commons license. Did Bogart choose the GFDL? I've forgotten what happened.

**Comment by oscarlevin** [comment]
For dimensions, right now I set up 7x10, but I would be happy to go to 8x10, leaving the text width the same, just increasing the outer margin for notes.  I don't think it make much of a difference cost wise to save 20 pages (last I checked, we could sell this thing for around $7 no matter what).

As for the license, I suppose we (and I really mean @mitchkeller) could try to contact the Bogart family and see if they would consider switching to Creative Commons.  This would be very nice for me, as I could see eventually creating a mashup of my book and this one, and I prefer the CC license (of course, I could give myself permission to license my book under GFDL, but would rather not).  

On the problem box decorations, I just pushed an update that looks better.  I'm still definitely open to other ideas.  As for the footnotes, I didn't find a nice solution, but now I'm thinking I prefer the "local" footnotes that are imposed by using a tcolorbox.  Any reason this is not okay?

**Comment by mitchkeller** [comment]
Yeah, I think that it will be plenty cheap regardless of size/page count. 7x10 seems reasonable to me.

As best as I can tell, there is an obligation to include the license. (In fact, https://wiki.creativecommons.org/wiki/GFDL_versus_CC-by-sa makes the point that for short works, the GFDL may be longer than the work itself, which can be annoying.) However, one idea I had (but didn't get a chance to comment on because my Christmas party was last night, and so I was baking up a storm all weekend) is that we could edit the LaTeX output to reduce the font size for that section. Even if doing so doesn't save us many pages, I still like that idea personally, since it emphasizes that the license is the fine print that we're required to include. The Bogart family (or someone at Dartmouth working with them) chose the GFDL. I agree it might be nice to have them issue a CC-BY-SA version, but I think we're unlikely to make progress on getting a license change before we want this print edition out. I'm happy to send the emails and work on that in January, however.

I'll review and merge @oscarlevin's layout tweaks (and hint markers) when I'm back on my other computer a bit later this afternoon. I have a draft cover, but I don't really care for it, so I'm working on some different ideas.

I actually don't mind the footnotes being "local" to the problems. The HTML version essentially behaves that way because of knowls. We're going to have a numbering inconsistency with the official Bogart PDF in terms of footnote numbering, but I think as long as we note that in our preface, it's fine. Would it be possible to change the footnote marker within the tcolorbox environments? I was thinking perhaps the standard symbols (dagger, etc.) for footnotes might be preferable to the lowercase letters in use now. If that's a pain to change, I don't think we need to muck with it.

**Comment by mitchkeller** [comment]
Here's a proposed cover. I have very few graphic design skills, but I did use a color scheme generator online to try to come up with some colors that went together well. Feel free to tell me you hate it or suggest some ways to improve it.

If we wanted to take a tiny bit of Bogart's preface and put it on the back, that might not be a bad idea as a way to give browsers at something like JMM some information to read without having to delve into the book's interior.
[ctgd_220_cover.pdf](https://github.com/OpenDiscreteMath/ibl-combinatorics/files/1552958/ctgd_220_cover.pdf)

**Comment by oscarlevin** [comment]
I like the cover.  Maybe move the title a little down (vertically centered in the orange)?  I like the idea of a blurb on the back.

**Comment by mitchkeller** [comment]
Here's a version that moves the title down. I didn't worry about precisely centering it, since trim irregularities can complicate that, but I think it's "morally centered" at least. I think I like this version better than my first one.
[ctgd_220_cover.pdf](https://github.com/OpenDiscreteMath/ibl-combinatorics/files/1553067/ctgd_220_cover.pdf)

**Comment by kentemorrison** [comment]
I like the cover and color scheme. The title moved down is better. Could you do a version using the same font as the cover of "Applied Combinatorics"? I think that typeface is quite pleasing. 

I agree that now is not the time to worry about the license.

**Comment by mitchkeller** [comment]
Here's a version with the font used for the cover on Applied Combinatorics.

[7x10_BW_220_bangla.pdf](https://github.com/OpenDiscreteMath/ibl-combinatorics/files/1553137/7x10_BW_220_bangla.pdf)

**Comment by oscarlevin** [comment]
I prefer the original font (especially the italic author name) and the lowered title.  The new font is okay, but maybe make it smaller if you want to go with that?

I assume the blue spine wraps around tot he front and back cover, so that the front cover would have a blue vertical rectangle on the left hand side, right?  

Maybe push the author name up and left a bit, in case of variations in trim (which CreateSpace creates a lot of!).

**Comment by kentemorrison** [comment]
Yes, I think the original serif font with italic author name does look better. (Mitch, Thanks for the extra work that turned out not to be necessary.)

**Comment by mitchkeller** [comment]
OK, good, we're unanimous on font then. I'm not an Illustrator power user, but I adept enough at it (and keeping old versions of files around) that it wasn't a big deal to change.

Here's a screenshot with the guides I used in Illustrator. The cyan lines are where the CreateSpace template puts the red "don't put things you don't want to risk getting trimmed in this region", not the idealized trim line locations. I feel like I've steered well clear enough of those, but I can nudge the author name a bit more if desired.

<img width="974" alt="bogart-cover-guides" src="https://user-images.githubusercontent.com/8580151/33913055-3a7c7090-df66-11e7-885a-677e61462d84.png">

**Comment by oscarlevin** [comment]
Awesome.  Thanks for all your work on this.

**Comment by kentemorrison** [comment]
Beautiful!

**Comment by mitchkeller** [comment]
Great. Sorry that the screenshot goes to the version with the alternate font. The version we'll use will have the serif font I proposed originally, as that's my main file now. Spacing from guides is the same in either case.

I like the new boxes, and I think the hint indicator is in as good of a place as it can be put. right now, if a problem has both a hint and an additional hint, we're getting two (h)'s. If it's a pain to avoid repeating, then leave it as is, but if it's an easy fix, then maybe it's better to have only one indicator? Hyperlinks would be nice, but they're not strictly necessary before going to print, since our goal isn't really to make the PDF version super awesome since we have the HTML.

I will have a PR in a bit with an amendment to our preface regarding the hint indicator and the footnote numbering.

What's the plan in terms of URL for where things will be living? We should have that in the print edition for sure.

**Comment by mitchkeller** [comment]
Realized that Oscar had already nicely set up bogart.openmathbooks.org to host all this, so I've dealt with URL issues both in the colophon and the preface.

It looks like that resolves the last of our issues before we can start the POD process. I guess Oscar and I need to decide which of us is setting up the book on CreateSpace or if we want to create a completely separate account to do so. I'm going to start checking overfull hboxes and such in the PDF output, which should finalize the page count and then allow me to lock down the cover.

**Comment by oscarlevin** [comment]
I'm happy to act as "publisher" and incur the initial costs of printing a few copies for JMM.  But if you, Mitch, would rather be in charge of this, that is fine too.  In some ways it makes sense for me to do it because I'm also hosting the html version (although we are talking $15 a year for the domain, and that's split with my own book).

**Comment by mitchkeller** [comment]
My preference was going to be for you to serve as the "publisher" for the reason you specified, so I say go for it.

The overfull hbox situation is really good. (Only two to fix.), so after I go to the gym, I should be able to send a commit that has a cover and everything ready for you to start battling with CreateSpace. I'm assuming that you've already made sure that the margins we're using are OK for their specifications?

**Comment by oscarlevin** [comment]
Sounds like a plan.  I assume the margins should be fine, given they are bigger than those I used on my book, which generally worked okay.

If you or @kentemorrison would like a print copy before JMM, I can ship them directly to you.  Or let me know how many you would like me to bring (beyond the display ones) and I can do so.

**Comment by kentemorrison** [comment]
Oscar and Mitch,

I’m busy today and so I’ve just seen the recent comments on publishing. Let me take care of the cost of books for JMM and copies for us. 

Kent

**Comment by mitchkeller** [comment]
I'd like one print copy, but it'd be fine if I got it at JMM, since that will save on shipping costs. I'll let the two of you work out payment and shipping arrangements.

**Comment by oscarlevin** [comment]
I've submitted the book to createspace, and it should be available in 24 hours.  However, I realized after submitting it that we had discussed putting a blurb on the back cover.  How about this, which I stole from the preface and am using as the book's description: 

This book is an introduction to combinatorial mathematics, also known as combinatorics. The book focuses especially but not exclusively on the part of combinatorics that mathematicians refer to as “counting.” The book consist almost entirely of problems. Some of the problems are designed to lead you to think about a concept, others are designed to help you figure out a concept and state a theorem about it, while still others ask you to prove the theorem. Other problems give you a chance to use a theorem you have proved. From time to time there is a discussion that
pulls together some of the things you have learned or introduces a new idea for you to work with. Many of the problems are designed to build up your intuition for how combinatorial mathematics works.

Above all, this book is dedicated to the principle that doing mathematics is fun. As long as you know that some of the problems are going to require more than one attempt before you hit on the main idea, you can relax and enjoy your successes, knowing that as you work more and more problems and share more and more ideas, problems that seemed intractable at first become a source of satisfaction later on.

@mitchkeller , would you be willing to add this, or some variant, to the amazon print cover?  I can resubmit that relatively easily.

**Comment by kentemorrison** [comment]
Sounds good to me!

**Comment by oscarlevin** [comment]
One more consideration: Createspace allows for "expanded distribution" which allows the book to be sold at other book stores and to libraries, etc.  There is no cost to us for this, but it does require that the book be priced a little higher.  With expanded distribution, we would price the book at 8.99, but without it we could go down to 6.99.  Any preference?

**Comment by kentemorrison** [comment]
Interesting question—not something I have run into before. The higher price might allow campus bookstores to sell it—if they’re even willing to stock a low priced book. Students pay a higher price in the bookstore but they don’t pay shipping. Even with sales tax it would be less than $10.

**Comment by mitchkeller** [comment]
New cover with blurb in #126. I added a sentence saying that the book is open-source and giving the URL, but otherwise went with the suggested text. I thought about the blurb after I sent my earlier PR but while I was off doing something else. It's probably a tiny bit better this way, anyway, since now the description on Amazon and the jacket blurb will match.

I vote yes for expanded distribution. I price my book that way so that bookstores can acquire them, and we have sold a few that way. When I teach calculus from Active Calculus, the campus store here actually sells a lot of copies of it off the shelf to students. For a student who has financial aid that covers their books, buying through the campus bookstore may be the only way they can use their financial aid to buy the book. Yes, that's more important for an expensive commercial text, but I still prefer to have as many options for buying as possible.

**Comment by oscarlevin** [comment]
Print edition is now approved through CreateSpace.  It should show up on Amazon in a few days.
